compiler/src/main/java/org/qbicc/type/CompoundType.java

```diff
@@ -258,6 +259,7 @@ public Builder setOverallAlignment(int align) {
         }
 
         public Builder addNextMember(final ValueType type) {
+            assert this.tag.equals(Tag.INLINE);
```

Comment:
These asserts will only fire if running java with `-ea` (enable assertions).  I think this would be better done either as an `io.smallrye.common.constraint.Assert`.  Applies above as well

Reply:
Thanks. I've made that change.